fearless/Modules/WalletMainContainer/WalletMainContainerPresenter.swift

```diff
@@ -152,3 +159,42 @@ extension WalletMainContainerPresenter: SelectNetworkDelegate {
         interactor.saveChainIdForFilter(chain?.chainId)
     }
 }
+
+extension WalletMainContainerPresenter: ScanQRModuleOutput {
+    func didFinishWith(addressInfo: AddressQRInfo) {
+        sendPrepareUseCase.getPossibleChains(
+            for: addressInfo.address,
+            delegate: self
+        )
+    }
+
+    func didFinishWith(address _: String) {}
+}
+
+extension WalletMainContainerPresenter: SendPrepareUseCaseDelegate {
+    func didReceive(possibleChains: [ChainModel]) {
+        if possibleChains.isEmpty {
```

Comment:
```suggestion
        guard possibleChains.isNotEmpty else { return }
```
Thus there will be no investment else if
(optional)